src/main/java/com/google/sps/servlets/GroceryServlet.java

```diff
@@ -107,27 +110,37 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     }
 
     // creates arraylist and starts query
-    ArrayList<String> groceryList = new ArrayList<String>();
+    ArrayList<Grocery> groceryList = new ArrayList<>();
     long familyID = (long) userInfoEntity.getProperty(FAMILY_ID);
     Query groceryQuery = new Query(GROCERY)
       .setFilter(new Query.FilterPredicate(FAMILY_ID, Query.FilterOperator.EQUAL, familyID));   
     PreparedQuery familyGrocery= datastore.prepare(groceryQuery);
-    groceryList = checkGroceries(familyGrocery);   
+    groceryList = checkGroceries(familyGrocery, userEmail);   
 
     Gson gson = new Gson();
     response.setContentType("application/json;");
     response.getWriter().println(gson.toJson(groceryList));
     }
 
-  // returns items from the query that match the users familyID
-  private ArrayList<String> checkGroceries(PreparedQuery familyGrocery) {
-    ArrayList<String> groceryList = new ArrayList<String>();
+  // returns items from the query that match the users familyID in the Grocery object    
+  private ArrayList<Grocery> checkGroceries(PreparedQuery familyGrocery, String userEmail) {
+    boolean match;
+    long timestamp = System.currentTimeMillis();
+    System.out.println(timestamp);
+    ArrayList<Grocery> groceryList = new ArrayList<>();
     for (Entity entity : familyGrocery.asIterable()) {
-        String groceryOutput;
-            String groceryItem = (String) entity.getProperty("grocery");
-            String memberEmail = (String) entity.getProperty("assignEmail");
-            groceryOutput = groceryItem + " assigned to: " + memberEmail;
-            groceryList.add(groceryOutput);
+        String groceryItem = (String) entity.getProperty(GROCERY);
+        String assignEmail = (String) entity.getProperty("assignEmail");
+        long id = entity.getKey().getId();     
+        if(userEmail.equals(assignEmail)) {
+            match = true;
+        } else if(assignEmail.equals(" ")){
```

Comment:
nit: rather than using empty string or `" "` I would suggest using null as the nonexistent value.

Reply:
when I change this value to null all of the buttons where no one is assigned disappear

Reply:
There are a couple of things that could be going on here:

1. You need to make sure you change the javascript condition to match.

2. Any items already stored in the Datastore would need to be changed as well, since the comparison wouldn't trigger true for any groceries we've already stored (but that's fine, because there's no real production data yet).

For the null/nonexistent value, I think `null` is my preference, followed by `""` (empty string of 0 length).

Reply:
what do you mean by  "null/nonexistent value, I think null is my preference, followed by "" (empty string of 0 length)"

Reply:
Essentially, what you're trying to represent with the `" "` here is the absence of a value ("no email address is assigned").

There are several common ways to represent absence of value. Using `null` to represent the absence of a value is common; using an empty string is also fine. (And also there are some more, if you want to do more reading, e.g. java.util.Optional: https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html)

So my ranking of preferences for how to represent "there is no email address assigned` is `null` -> "" (the empty string) -> " " (which you are using here).